src/main/java/ca/corefacility/bioinformatics/irida/ria/web/pipelines/PipelineController.java

```diff
@@ -195,25 +178,37 @@ public String getPipelineLaunchPage(final Model model, Locale locale) {
 
 	/**
 	 * Get a generic pipeline page.
-	 * 
-	 * @param model
-	 *            the the model for the current request
-	 * @param principal
-	 *            the user in the current request
-	 * @param locale
-	 *            the locale that the user is using
-	 * @param pipelineId
-	 *            the pipeline to load
+	 *
+	 * @param model      the the model for the current request
+	 * @param principal  the user in the current request
+	 * @param locale     the locale that the user is using
+	 * @param pipelineId the pipeline to load
+	 * @param projectId  Project ID to add a {@link ca.corefacility.bioinformatics.irida.model.workflow.submission.AnalysisSubmissionTemplate}
+	 *                   if necessary (not required)
 	 * @return a page reference or redirect to load.
```

Comment:
I know it should not be part of this merge request, but this controller is **MASSIVE**.  It is currently at 230 lines.  We need to find a better way of creating this page.

Reply:
Agreed.  A lot of the analysis code has got a bit out of hand.  At the very least this could be broken down into a handful of private methods so that the main controller method is easier to read.  This is one of the refactor targets we need to look at anyway so I'd say leave it for a later full refactor merge.  See #126